### DISCUSSION TITLE
WinError 10048 on some device

### DISCUSSION THREAD

**Discussion by marcowg** [jasonacox/tinytuya#539] [General]
When using a Outletdevice, now and then they stop working and give WinError 10048 on the socket as result.
the cloud functions keep on working normal. 
A power cycle is the only way to solve it.

Any ideas?

**Comment by uzlonewolf** [comment]
Let me guess, it's a v3.4 device?  This is a known problem with them and the only way to fix it is to replace it with a v3.3 or v3.5 one.

**Comment by marcowg** [comment]
in scan it report as 3.3 and I use d.set_version(3.3).

Any ideas?

**Reply by jasonacox** [reply]
When it stops responding, do you still see it in a scan? Also, do you have other devices that work fine?

**Reply by marcowg** [reply]
Yes it still comes in the scan. I have +- 35 device working fine , but 2 often run in to this problem.

**Reply by jasonacox** [reply]
Can you post the error you get? Also, just to eliminate all possibilities, do you have a 2nd computer you can try to access these devices during this state?  Super long shot, but could be something network or arp related. However, my bet is still on these having a bug in their firmware that crashes the local API after an amount of use (think memory leak).  If that, my solution has been to replace them.

**Reply by marcowg** [reply]
I can not connect from other systems. 
I saw thery have lower firmware versions in the smartlife app , but the are not updatable.
so I started a call with smartlife , thx soo far.
